examples/medplum-chat-demo/src/components/actions/AddParticipant.tsx

```diff
@@ -0,0 +1,129 @@
+import { Button, Modal } from '@mantine/core';
+import { useDisclosure } from '@mantine/hooks';
+import { showNotification } from '@mantine/notifications';
+import { normalizeErrorString, PatchOperation } from '@medplum/core';
+import { Communication, Questionnaire, QuestionnaireResponse } from '@medplum/fhirtypes';
+import { QuestionnaireForm, useMedplum } from '@medplum/react';
+import { IconCircleCheck, IconCircleOff } from '@tabler/icons-react';
+import { checkForInvalidRecipient, getRecipients } from '../../utils';
+
+interface AddParticipantProps {
+  readonly communication: Communication;
+  readonly onChange: (communication: Communication) => void;
+}
+
+export function AddParticipant(props: AddParticipantProps): JSX.Element {
+  const medplum = useMedplum();
+  const [opened, handlers] = useDisclosure(false);
+
+  const onQuestionnaireSubmit = (formData: QuestionnaireResponse): void => {
+    const newParticipantsData = getRecipients(formData);
+    if (!newParticipantsData) {
+      throw new Error('Please select a valid person to add to this thread.');
+    }
+    const newParticipants = newParticipantsData?.map(
+      (participant) => participant.valueReference
+    ) as Communication['recipient'];
+
+    if (!newParticipants) {
+      throw new Error('Please select a valid person to add to this thread.');
+    }
+
+    const invalidRecipients = checkForInvalidRecipient(newParticipants);
+
+    if (invalidRecipients) {
+      showNotification({
+        color: 'red',
+        icon: <IconCircleOff />,
+        title: 'Error',
+        message: 'Invalid recipient type',
+      });
+      throw new Error('Invalid recipient type');
+    }
+
+    handleNewParticipant(newParticipants).catch(console.error);
+    handlers.close();
+  };
+
+  const handleNewParticipant = async (newParticipant: Communication['recipient']): Promise<void> => {
```

Comment:
rename to `addNewParticipant` for clarity. 

Also prefer declaration as `async function addNewParticipant(...){...}`